lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -468,8 +472,48 @@ def add_tax_service(xml)
         end
       end
 
-      def add_auth_service(xml)
-        xml.tag! 'ccAuthService', {'run' => 'true'}
+      def add_auth_service(xml, payment_method, options)
+        if network_tokenization?(payment_method)
+          add_network_tokenization(xml, payment_method, options)
+        else
+          xml.tag! 'ccAuthService', {'run' => 'true'}
+        end
+      end
+
+      def network_tokenization?(payment_method)
+        payment_method.is_a?(NetworkTokenizationCreditCard)
+      end
+
+      def add_network_tokenization(xml, payment_method, options)
+        return unless network_tokenization?(payment_method)
```

Comment:
Do you need this check? It seems like you're only calling `add_network_tokenization` in situations where you already know that the `payment_method` is a `NetworkTokenizationCreditCard`.